packages/config/src/normalize.js

```diff
@@ -1,15 +1,24 @@
 'use strict'
 
-const { normalize: normalizeFunctionsConfig } = require('./functions_config')
+const {
+  normalize: normalizeFunctionsConfig,
+  WILDCARD_ALL: FUNCTIONS_CONFIG_WILDCARD_ALL,
+} = require('./functions_config')
 const { deepMerge } = require('./utils/merge')
 const { removeFalsy } = require('./utils/remove_falsy')
 
 // Normalize configuration object
 const normalizeConfig = function (config) {
   const { build, functions, plugins, ...configA } = deepMerge(DEFAULT_CONFIG, config)
   const functionsA = normalizeFunctionsConfig(functions)
+  const topLevelFunctions = functionsA[FUNCTIONS_CONFIG_WILDCARD_ALL] || {}
+  const buildA = {
+    ...build,
+    functions: topLevelFunctions.directory || build.functions,
+  }
   const pluginsA = plugins.map(normalizePlugin)
-  return { ...configA, build, functions: functionsA, plugins: pluginsA }
+
+  return { ...configA, build: buildA, functions: functionsA, plugins: pluginsA }
 }
 
 const DEFAULT_CONFIG = {
```

Comment:
Should we change `functions: {}` to `functions: { [FUNCTIONS_CONFIG_WILDCARD_ALL]: {} }` in `DEFAULT_CONFIG`? This wold remove the need for `functionsA[FUNCTIONS_CONFIG_WILDCARD_ALL] || {}`